### ISSUE TITLE
client 2.0 (the one with ethers.js support) is not working with infura

### ISSUE THREAD

**Issue by jellegerbrandy** [daostack/arc.js#445]
There are two reasons here:

(a) The rpc provider does not seem to work well. One workaround:

```
const infuraProvider = new ethers.providers.InfuraProvider('rinkeby', 'e0cdf3bfda9b468fa908aa6ab03d5ba2')
    arc.web3 = infuraProvider
```

(b) the `arc.getContract(..)`  function uses `getSigner` in the ethereum provider, which is not supported by infura. We need to provide the wallet ourselves here.

**Comment by dOrgJelli** [comment]
Can you describe the use case @jellegerbrandy? If this is in the browser, and let's say metamask is in use, we should be constructing arc.ts with an instance of metamask. I just pushed a fix to allow for this here: https://github.com/daostack/client/pull/446

If instead you're looking to sign things locally with an in-memory private key, and then use an Infura provider, then we'll have to modify arc.ts to support the user providing an ethers Wallet instance in addition to the web3 provider. Is this required?

**Comment by jellegerbrandy** [comment]
that is not the point :-)

the point is that

```arc = new Arc( ... use infurarpovers...)
arc.getContract(address, abi)
```
will trhow an error
```
{ Error: INFURA does not support signing (operation="getSigner", version=4.0.46)
    at Object.throwError (/home/jelle/projects/daostack/common-scripts/node_modules/ethers/errors.js:76:17)
    at InfuraProvider.getSigner (/home/jelle/projects/daostack/common-scripts/node_modules/ethers/providers/infura-provider.js:77:23)
    at Arc.getContract (/home/jelle/projects/daostack/common-scripts/node_modules/@daostack/client-experimental/dist/lib/arc.js:314:62)
    at triggerEvent (/home/jelle/projects/daostack/common-scripts/index.js:60:30)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
  reason: 'INFURA does not support signing',
  code: 'UNSUPPORTED_OPERATION',
  operation: 'getSigner' }
```
from this line: https://github.com/daostack/client/blob/client-2-0/src/arc.ts#L327 (or similar)

**Comment by dkent600** [comment]
Why does `getContract` need a signier?

**Comment by dOrgJelli** [comment]
@dkent600 the `Contract` constructor requires a signerOrProvider: https://github.com/ethers-io/ethers.js/blob/b1c6575a1b8cc666a9173eceedb7a367329819c7/src.ts/contract.ts#L369

@jellegerbrandy we could add a work around for this by changing the logic [here](https://github.com/daostack/client/blob/629c118f26e1f9c117c5f126dfb2a1e3a7aed49d/src/arc.ts#L327) to be:
```typescript
try {
  // case 1
  return new Contract(address, abi, this.web3.getSigner(this.defaultAccount))
} catch (err) {
  // case 2
  return new Contract(address, abi, this.web3)
}
```

In `// case 2` if you try to send transactions on the contract instance, it'll result with an exception here: https://github.com/ethers-io/ethers.js/blob/b1c6575a1b8cc666a9173eceedb7a367329819c7/src.ts/contract.ts#L264

**Comment by dkent600** [comment]
`signerOrProvider` can be just a provider.  doesn't need to be a signer

**Comment by dkent600** [comment]
can you mock a signer?

**Comment by dOrgJelli** [comment]
@dkent600 if we want to send transactions on the contract and not just calls, we'll need a valid signer.

@jellegerbrandy can you specify if this is your use-case:
- using Infura for the RPC connection
- using an in-app-memory private key to sign transactions

If this is the case, we can change the interface for Arc to accept a `Wallet` in addition to a `Web3Provider`.

**Comment by jellegerbrandy** [comment]
adding a wallet in the config would be good for my use case...

**Comment by dOrgJelli** [comment]
Okay, so you have either a private key or a mnemonic you'd like to use @jellegerbrandy?

**Comment by jellegerbrandy** [comment]
yes!
Basically, I want to do something like this:

```
const infuraProvider = new ethers.providers.InfuraProvider('rinkeby', 'e0cdf3bfda9b468fa908aa6ab03d5ba2')
const wallet = new ethers.Wallet(PRIVATE_KEY_1, infuraProvider)

const arc = new Arc({ .. web3Provider: wallet })
````

**Comment by dOrgJelli** [comment]
Amazing, we can certainly support this.

**Comment by jellegerbrandy** [comment]
was fixed with #450
